piezo_web_app/PiezoWebApp/src/services/spark_job/spark_job_service.py

```diff
@@ -271,3 +279,43 @@ def _retrieve_status(item):
         else:
             self._logger.debug(f'Not processing job "{job}", current status is "{status}"')
             return TidiedJobStatus(job, status, "NO", None, None)
+
+    def _expose_spark_ui(self, job_name):
```

Comment:
`_expose_spark_ui` and `_delete_spark_ui_components` feel like they should belong in the `SparkUiService`. It _might_ then be sensible to move some of what is currently in `SparkUiService` to a `SparkUiAdapter` (or similar).